File: insurance_backend/src/main/java/com/maghrebia/data_extract/DTO/CreateProductionDTO.java
package com.maghrebia.data_extract.DTO;

import java.util.Date;
import java.util.List;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@AllArgsConstructor
@NoArgsConstructor
@Data
public class CreateProductionDTO {

    private String nature;	
    private Integer codeRisque;		
    private Date dateEffet;	
    private Date dateEcheance;
    private Integer mois;
    private String dureeContrat;	
    private String modePayement;		
    private Integer nombreCheque;		
    private String numeroCheque;	
    private Date dateDuCheque;
    private float primeNette; 	
    private float prime;		
    private float commission;
    private String remarques;

    private List<CreateBanqueDTO> transactions;

}

File: insurance_frontend/src/app/models/create-production.dto.ts
export interface CreateProductionDTO {
    idProduction: number;
    numeroContrat: string;
    nature: string;
    dateEffet: Date;
    dateEcheance: Date;
    mois: number;
    dureeContrat: number;
    modePayement: string;
    nombreCheque: number;
    numeroCheque: string;
    dateDuCheque: Date;
    primeNette: number;
    prime: number;
    commission: number;
    remarques: string;
  }
  